main.nf

```diff
@@ -82,22 +100,43 @@ log.info """\
         qualimap_version: ${params.qualimap_version}
         bamqc_outformat: ${params.bamqc_outformat}
         bamqc_additional_options: ${params.bamqc_additional_options}
+
+    - FastQC options:
+        fastqc_version: ${params.fastqc_version}
+        fastqc_additional_options: ${params.fastqc_additional_options}
 """
 
 Channel
     .fromList(params.samples_to_process)
-    .map { sample ->
-        return tuple(sample.orig_id, sample.id, sample.path, sample.read_length, sample.sample_type)
+    .map { sm ->
+        def rg_arg = ""
+        return tuple(sm.path, sm.orig_id, sm.sm_id, rg_arg, null, null, sm.sample_type, sm.read_length)
     }
-    .set { samplesToProcessChannel }
+    .set { samples_to_process_ch }
+
+Channel
+    .fromList(params.libraries_to_process)
+    .map { lib ->
+        def rg_arg = lib.rgs.collect { "-r ${it}" }.join(' ')
```

Comment:
I forgot to change this.  In a separate PR I will make this a list of readgroups and add the `-r` within the  processes

Reply:
I also think I should impose a maximum number of readgroups, and I suppose libraries, that will be processed separately.  Some of the GIAB samples for example have > 1000 readgroups.

Reply:
I added parameters for maximum number of readgroups and libraries to process separately with defaults = 20.

I also parameterized a choice of levels to run `FastQC`.